packages/arb-token-bridge-ui/src/components/syncers/PendingTransactionsUpdater.tsx

```diff
@@ -27,7 +33,21 @@ export function PendingTransactionsUpdater(): JSX.Element {
         return null
       }
 
-      return provider.getTransactionReceipt(tx.txID)
+      if (tx.type == 'deposit-l1') {
+        // We need to get the seqNum for deposit tx if its missing
+        return provider
+          .getTransactionReceipt(tx.txID)
+          .then(async (txr: TransactionReceiptWithSeqNum) => {
+            const l1ToL2Msg = await new L1TransactionReceipt(
+              txr
+            ).getL1ToL2Message(provider)
+            const seqNum = l1ToL2Msg.messageNumber
+            txr.seqNum = seqNum.toNumber()
+            return txr
+          })
```

Comment:
Two tiny things I'd change:

* return value of `getTransactionReceipt` should be `TransactionReceipt`
* return a fresh object spreading over properties from `txr` instead of mutating it

```
return provider
  .getTransactionReceipt(tx.txID)
  .then(async (txr: TransactionReceipt) => {
    const l1ToL2Msg = await new L1TransactionReceipt(txr).getL1ToL2Message(provider)
    return { ...txr, seqNum: l1ToL2Msg.messageNumber.toNumber()
  })
```

Reply:
Thanks that looks much cleaner!